apps/common/components/Input.tsx

```diff
@@ -0,0 +1,53 @@
+import type {ReactElement} from 'react';
+
+type TAmountInputProps = {
+	value: string;
+	onChange?: (value: string) => void;
+	label?: string;
+	placeholder?: string;
+	legend?: string;
+	error?: string;
+	disabled?: boolean;
+	loading?: boolean;
+}
+
+function Input({
+	value,
+	onChange,
+	label,
+	placeholder,
+	legend,
+	error,
+	disabled,
+	loading
+}: TAmountInputProps): ReactElement {
+	return (
+		<div className={'w-full'}>
+			{label && (
+				<p
+					className={'mb-1 w-full overflow-hidden text-ellipsis whitespace-nowrap text-base text-neutral-600'}>
+					{label}
+				</p>
+			)}
+			<div className={'relative flex w-full items-center justify-center'}>
+				<input
+					className={`h-10 w-full p-2 font-mono text-base font-normal outline-none ${error ? 'border border-solid border-[#EA5204] focus:border-[#EA5204]' : 'border-0 border-none'} ${disabled ? 'bg-neutral-300 text-neutral-600' : 'bg-neutral-0'}`}
```

Comment:
Do you know if there's a better way to handle these states rather than having all these ternary ops scattered around the `className` string? Looks hard to read, if there would be a better way that would be great :)